src/meatie_aiohttp/client.py

```diff
@@ -32,6 +33,10 @@ def __init__(
         self.session = session
         self.session_params = session_params if session_params else {}
 
+    @classmethod
+    def loads(cls) -> JSONDecoder:
+        return DEFAULT_JSON_DECODER
```

Comment:
Could you modify the way the decoder is being handled? The current approach with a class method requires users who would like to change the DEFAULT_JSON_DECODER to use inheritance.

Instead, could you add the `loads` argument to constructor with the default decoder used by aiohttp? This way it will hopefully simplify the change of the default serializer if needed. It should also make it more clear, since the argument will be present in the constructor.

Reply:
I do not think `loads` should be a constructor parameter. The decision on how JSON should be parsed is the client creator's decision, not the client user's. 
Changing the `loads` might make returning types from methods wrong or broken.

It's possible to introduce metaclass and add `loads` as a constrictor parameter for `meatie_*.Client` class:
```
from meatie_aiohttp import Client

class MyClient(Client, loads=custom_loads):
   ....
```

but this solution confuses library users much more than adding class method and ability to override it

Reply:
Hi Damian,

I revised carefully how the mainstream Python HTTP client libraries (aiohttp, requests, and xhttp) handle the deserialization of JSON or text in the HTTP response body.

I summarised it below:

`aiohttp` supports replacing `json.loads` and encoding
`httpx` supports passing additional parameters to `json.loads`
`requests` support passing additional parameters to `json.loads` and encoding

Consequently, every library takes a slightly different approach. Only `aiohttp` library allows using a custom JSON decoder that doesn't come from the Python standard library.

Aiming to keep Meatie independent of the HTTP client library I introduced the `body` parameter that allows specifying a custom function that extracts JSON from the HTTP response (https://github.com/pmateusz/meatie/pull/28). It enables custom JSON processing on a per-endpoint basis. The analogous mechanism is implemented for extracting `text`.  I added this feature for all HTTP clients supported by Meatie.

Essentially, the client developer decides how JSON should be processed, which is aligned with what you postulate.

As a digression, I was experimenting with metaclasses before the first public release of the Meatie. Metaclasses were poorly supported by mypy and the IntelliJ code checker. Almost every method of a Meatie client was highlighted incorrectly as having a wrong type. Therefore, I decided to abandon it in favor of the current implementation.

Thank you for the feature proposal and implementation for `aiohttp`. I hope the final implementation supports your use case.